addons/account/report/account_tax_details_report.py

```diff
@@ -0,0 +1,385 @@
+                    WHERE base_tax_matching_no_tax_line_as_base.tax_line_id IS NULL
+
+                ),
+
+                base_tax_matching_all_amounts AS (
+
+                    SELECT
+                        sub.tax_line_id,
+                        sub.base_line_id,
+                        sub.src_line_id,
+
+                        tax_line.tax_line_id AS tax_id,
+                        tax_line.group_tax_id,
+
+                        tax_line.company_id,
+                        comp_curr.id AS company_currency_id,
+                        comp_curr.decimal_places AS comp_curr_prec,
+                        curr.id AS currency_id,
+                        curr.decimal_places AS curr_prec,
+
+                        sub.base_amount,
+                        SUM(
+                            CASE WHEN tax.amount_type = 'fixed'
+                            THEN CASE WHEN base_line.balance < 0 THEN -1 ELSE 1 END * ABS(COALESCE(base_line.quantity, 1.0))
+                            ELSE sub.base_amount
+                            END
+                        ) OVER (PARTITION BY tax_line.id ORDER BY tax_line.tax_line_id, sub.base_line_id, sub.src_line_id) AS cumulated_base_amount,
+                        SUM(
+                            CASE WHEN tax.amount_type = 'fixed'
+                            THEN CASE WHEN base_line.balance < 0 THEN -1 ELSE 1 END * ABS(COALESCE(base_line.quantity, 1.0))
+                            ELSE sub.base_amount
+                            END
+                        ) OVER (PARTITION BY tax_line.id) AS total_base_amount,
+                        tax_line.balance AS total_tax_amount,
+
+                        sub.base_amount_currency,
+                        SUM(
+                            CASE WHEN tax.amount_type = 'fixed'
+                            THEN CASE WHEN base_line.amount_currency < 0 THEN -1 ELSE 1 END * ABS(COALESCE(base_line.quantity, 1.0))
+                            ELSE sub.base_amount_currency
+                            END
+                        ) OVER (PARTITION BY tax_line.id ORDER BY tax_line.tax_line_id, sub.base_line_id, sub.src_line_id) AS cumulated_base_amount_currency,
+                        SUM(
+                            CASE WHEN tax.amount_type = 'fixed'
+                            THEN CASE WHEN base_line.amount_currency < 0 THEN -1 ELSE 1 END * ABS(COALESCE(base_line.quantity, 1.0))
+                            ELSE sub.base_amount_currency
+                            END
+                        ) OVER (PARTITION BY tax_line.id) AS total_base_amount_currency,
+                        tax_line.amount_currency AS total_tax_amount_currency
+
+                    FROM base_tax_matching_base_amounts sub
+                    JOIN account_move_line tax_line ON
+                        tax_line.id = sub.tax_line_id
+                    JOIN account_move_line base_line ON
+                        base_line.id = sub.base_line_id
```

Comment:
couldn't we fetch directly the fields we need here in subqueries, to avoid joining again with account_move_line?

Reply:
I tried it but for some reason, it was less efficient :/ To recheck